assets/js/webinar.js

```diff
@@ -8,6 +8,40 @@ function localizeDateTime(date) {
     return localeDate.toLocaleString(undefined, options);
 }
 
+/**
+ * This function parses the cookie string and returns an object of the current cookies.
+ */
+function parseCookie() {
+    var cookieObject = {};
+    var cookieString = document.cookie;
+    var parsedCookie = cookieString.split(";").map(function(cookie) {
+        var parts = cookie.split("=");
+        cookieObject[parts[0]] = parts[1];
+        return cookie;
+    });
+
+    return cookieObject
+}
+
+/**
+ * This function parses the UTM cookie and returns and object of the cookie.
+ *
+ * @param {string} utmCookieString The value of the '__utmzz' cookie. Values are separated by '|'.
+ */
+function parseUTMCookie(utmCookieString) {
+    var cookie = utmCookieString ? utmCookieString : "";
+    var parts = cookie.split("|");
+    var utmObj = {};
+
+    for (let i = 0; i < parts.length; i++) {
+        var value = parts[i];
+        var valueParts = value.split("=");
+        utmObj[valueParts[0]] = utmObj[1];
+    }
+
+    return parseUTMCookie;
```

Comment:
`utmObj[valueParts[0]] = utmObj[1];` doesn't look right.

How about something like this:

```suggestion
    return (utmCookieString || "").split("|").reduce(function(obj, utm) {
        var parts = utm.split("=");
        obj[parts[0]] = parts[1];
        return obj;
    }, {});
```

Reply:
👍